src/client.ts

```diff
@@ -523,12 +523,11 @@ export class StreamChat<StreamChatGenerics extends ExtendableGenerics = DefaultG
       throw Error('User is not set on client, use client.connectUser or client.connectAnonymousUser instead');
     }
 
-    if (this.wsConnection?.isConnecting) {
+    if (this.wsConnection?.isConnecting && this.wsPromise) {
       this.logger('info', 'client:openConnection() - connection already in progress', {
         tags: ['connection', 'client'],
       });
-
-      return Promise.resolve();
+      return this.wsPromise;
```

Comment:
fix: If the connection is in progress, then we need to return the existing promise instead of resolving the promise.